### PR TITLE
Fix signaling events roomlist

### LINKED ISSUE #1810: Improve conversation list reload condition
Currently the internal and standalone signaling backends reload the conversation list every 30 seconds.

In an ideal world the reload would happen:
* When the user marked a conversation read in another device/tab
* When a message is posted in a conversation, which is not the conversation the user is looking at right now. Pseudo query:
    ```sql
    select p1.session_id
    from participants1
    left join participants2
      on p2.user_id = p1.user_id
    WHERE p1.session IS NOT NULL
      and p2.room = $updatedRoom
      and p1.room != $updatedRoom
      and p2.user_id IS NOT NULL
     ```

And after that there should be a grace period, so posting 5 messages does not update the list 5 times

### LINKED ISSUE #6706: Signaling event roomlist.update triggered too often?
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

It's been some time that we removed the participant information from the room list end point, so this might be caused by legacy, but:

## Steps to reproduce
1. As user 1 create a room with many users
1. As user 2 create a one-to-one with user 1
1. As user 1 start a call in the room with many users
1. As user 1 open the console / network tab
1. As user 2:
    i. Join the one-to-one
    i. Wait 3 seconds
    i. Join the shared call
    i. Wait 3 seconds
    i. Join the one-to-one
    i. Wait 3 seconds
    i. Join the shared call

### Expected behaviour
As user 1 there was no actual reason to reload your room list

### Actual behaviour
As user 1 your room list is refreshed for each of the joins,
But none of the values in there actually changed
```json
{
  "type": "event",
  "event": {
    "target": "roomlist",
    "type": "update",
    "update": {
      "roomid": "m6turkvr",
      "properties": {
        "name": "Private conversation",
        "type": 2,
        "lobby-state": 0,
        "lobby-timer": null,
        "read-only": 0,
        "listable": 0,
        "active-since": {
          "date": "2021-12-14 15:07:05.000000",
          "timezone_type": 3,
          "timezone": "UTC"
        },
        "sip-enabled": 0
      }
    }
  }
}
{
  "type": "event",
  "event": {
    "target": "roomlist",
    "type": "update",
    "update": {
      "roomid": "5878504764",
      "properties": {
        "name": "Private conversation",
        "type": 1,
        "lobby-state": 0,
        "lobby-timer": null,
        "read-only": 0,
        "listable": 0,
        "active-since": null,
        "sip-enabled": 0
      }
    }
  }
}
```

## Talk app

**Talk app version:** master and 13.0.1

**Custom Signaling server configured:** yes

**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #6706 
Fix #1810

### CODE REVIEW

**Review by danxuliu — COMMENTED** [review]
Besides the mentioned issue, tested and works.

I think it would be good to clarify in the [API documentation](https://github.com/nextcloud/spreed/blob/deb2494193a51321273070b39739e625651d40ea/docs/standalone-signaling-api-v1.md#room-list-events) that `roomlist` events with type `invite` or `disinvite` are sent only to the users being invited/disinvited, while `roomlist` events with type `update` are sent to all the users in the room that was modified (specified in the `roomid` property). Also describing the `participant-list` property would be good too.

File: src/utils/signaling.js [hunk 1]
```diff
@@ -1219,8 +1219,19 @@ Signaling.Standalone.prototype.processRoomMessageEvent = function(data) {
 
 Signaling.Standalone.prototype.processRoomListEvent = function(data) {
 	switch (data.event.type) {
+	case 'update':
+		if (data.event.update.properties['participant-list']) {
+			console.debug('Room list event for participant list', data)
+			if (data.event.update.roomid === this.currentRoomToken) {
+				// This is handled via the join/leave events already
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/utils/signaling.js]: RIght, fixed with e985ce67f9bd813bd042f26c841ea16df5d6677f

**Review by danxuliu — APPROVED** [review]
Tested and works :+1:

**Pull request merged**